ec/ecresource/deploymentresource/elasticsearch_expanders.go

```diff
@@ -174,6 +183,22 @@ func expandEsConfig(raw interface{}, esCfg *models.ElasticsearchConfiguration) e
 	return nil
 }
 
+func expandSnapshotSource(raw interface{}, restore *models.RestoreSnapshotConfiguration) error {
```

Comment:
This function never returns any error, maybe a good idea to just remove the `error` return from the signature?